### PR TITLE
Add `useRedirectLoggedInUserTo()` hook

### PR DESCRIPTION
**Pull request by engelkes-finstreet**
Some users were asking how to redirect from login page if user is already logged in. Small change to redirect to the root page if a users is logged in from `signup` / `login` page.

### CODE REVIEW

File: packages/generator/templates/app/app/hooks/redirectLoggedInUser.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { useRouter, useSession } from "@blitzjs/core"
+import { useEffect } from "react"
+
+export const useRedirectLoggedInUser = (url: string) => {
+  const session = useSession()
+  const router = useRouter()
+
+  useEffect(() => {
+    if (session.userId) {
+      router.push(url, url)
```
Comment (flybayer) [inline packages/generator/templates/app/app/hooks/redirectLoggedInUser.ts:10]:
```suggestion
      router.push(url)
```

File: packages/generator/templates/app/app/auth/pages/signup.tsx [hunk 2]
```diff
@@ -2,8 +2,10 @@ import React from "react"
 import { useRouter, BlitzPage } from "blitz"
 import Layout from "app/layouts/Layout"
 import { SignupForm } from "app/auth/components/SignupForm"
+import { useRedirectLoggedInUser } from "../../hooks/redirectLoggedInUser"
```
Comment (flybayer) [inline packages/generator/templates/app/app/auth/pages/signup.tsx:5]:
```suggestion
import { useRedirectLoggedInUser } from "app/hooks/redirectLoggedInUser"
```

File: packages/generator/templates/app/app/auth/pages/login.tsx [hunk 3]
```diff
@@ -2,8 +2,10 @@ import React from "react"
 import { useRouter, BlitzPage } from "blitz"
 import Layout from "app/layouts/Layout"
 import { LoginForm } from "app/auth/components/LoginForm"
+import { useRedirectLoggedInUser } from "../../hooks/redirectLoggedInUser"
```
Comment (flybayer) [inline packages/generator/templates/app/app/auth/pages/login.tsx:5]:
```suggestion
import { useRedirectLoggedInUser } from "app/hooks/redirectLoggedInUser"
```

**Review by flybayer — COMMENTED** [review]
Awesome, thanks!!

What do you think about changing name from `redirectLoggedInUser()` to `redirectLoggedInUserTo()`? I feel like the later reads a bit better and is more clear about what the function argument is for.

**Comment by engelkes-finstreet** [comment]
> Awesome, thanks!!
> 
> What do you think about changing name from `redirectLoggedInUser()` to `redirectLoggedInUserTo()`? I feel like the later reads a bit better and is more clear about what the function argument is for.

Way better name 👍  You are right!

**Comment by flybayer** [comment]
Ack, I don't know why I didn't think about this before, but let's just add this hook to blitz/core! And then use that inside the templates.

**Comment by engelkes-finstreet** [comment]
I was thinking about this as well, but I do not like the flickering to be honest. To use this for the special case that a user loggen in it is fine but I think we should provide a better solution for different cases.

**Comment by engelkes-finstreet** [comment]
@flybayer To get this PR closed - Your call if I should move it inside core package or not.

**Comment by flybayer** [comment]
Well, since we aren't super happy with the UX of it, let's close this PR and instead add the hook source to the docs as an example for people. Later we can revisit adding something like it to core.


To: https://blitzjs.com/docs/authorization

**Pull request closed without merging**